sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,267 @@
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+    def fit_transform(self, X, y, **fit_params):
+        """Fit estimator and transform dataset.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data used to build forests. Use ``dtype=np.float32`` for
+            maximum efficiency.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
+            Transformed dataset.
+        """
+        preds = cross_val_predict(self.base_estimator, X, y, cv=self.cv,
+                                  method=self._method_name(),
+                                  n_jobs=self.n_jobs, fit_params=fit_params)
+
+        self.base_estimator.fit(X, y, **fit_params)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+
+def _identity(x):
+    return x
+
+
+def _identity_transformer():
+    """Contructs a transformer that does nothing"""
+    return FunctionTransformer(_identity)
+
+
+class StackLayer(FeatureUnion):
+    """ Single layer for model stacking
+
+    Parameters
+    ----------
+    base_estimators : list of estimators to be used in stacking
+
+    restack : bool, optional (default=False)
+        Whether input should be concatenated to the transformation.
+
+    cv : cv to be used, optional (default=3)
+        Will be passed to `StackMetaEstimator` for each base estimator.
+
+    method : string, optional (default='auto')
+        Invokes the passed method name of the estimators. If the method is
+        `auto`, will try to invoke `predict_proba` or `predict` in that
+        order.
+
+    n_jobs : int, optional (default=1)
+        Number of jobs to be passed to `cross_val_predict` during
+        `fit_transform`.
+
+    transformer_weights : dict, optional (default=None)
+        Multiplicative weights for features per transformer.
+        Keys are transformer names, values the weights.
+
+    Example
+    -------
+    >>> from sklearn.ensemble import StackLayer
+    >>> from sklearn.linear_model import Ridge
+    >>> from sklearn.svm import SVC
+    >>> l = StackLayer([('svc', SVC()), ('ridge', Ridge())])
+    >>> l # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+    StackLayer(base_estimators=[('svc', SVC(C=1.0, ...)),
+                                ('ridge', Ridge(alpha=1.0, ...))],
+               cv=3, method='auto', n_jobs=1, restack=False,
+               transformer_weights=None)
+    """
+    def __init__(self, base_estimators=[], restack=False, cv=3, method='auto',
+                 n_jobs=1, transformer_weights=None):
+        self.base_estimators = base_estimators
+        self.restack = restack
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
```

Comment:
If we are keeping the `StackLayer` class, I'd argue that it'd be better to leave the param "closer" to the outermost class. If a user wants to tweak `StackMetaEstimator` params, then it's easy to use that directly instead of the helper class. Does that make sense?